### PR TITLE
Use the new format for consuming VSC Proposed API

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -352,4 +352,6 @@ MigrationBackup/
 node_modules
 out/
 dist/
-.vscode-test/
\ No newline at end of file
+.vscode-test/
+vscode.d.ts
+vscode.proposed.*.d.ts
```
Comment (DonJayamanne) [inline .gitignore:357]: We don't need these in the repo

File: package.json [hunk 2]
```diff
@@ -14,10 +14,18 @@
...
         "webpack": "webpack --mode production && node ./scripts/optimizeTypings.js",
         "webpack-link": "webpack --mode development && node ./scripts/optimizeTypings.js",
-        "webpack-dev": "webpack --mode development --watch"
+        "webpack-dev": "webpack --mode development --watch",
+		"postinstall": "npm run download-api"
     },
     "author": "Visual Studio Code Team",
     "license": "MIT",
+    "engines": {
+        "vscode": "^1.64.0"
```
Comment (DonJayamanne) [inline package.json:23]: To ensure the right d.ts files are pulled down

File: package.json [hunk 3]
```diff
@@ -14,10 +14,18 @@
...
-        "webpack-dev": "webpack --mode development --watch"
+        "webpack-dev": "webpack --mode development --watch",
+        "postinstall": "npm run download-api"
     },
     "author": "Visual Studio Code Team",
     "license": "MIT",
+    "engines": {
+        "vscode": "^1.64.0"
+    },
+    "enabledApiProposals": [
```
Comment (rchiodo) [inline package.json:25]: Are these necessary for something that doesn't create an extension?

**Review by rchiodo — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (DonJayamanne) [reply package.json:25]: at the end of the day we are still using the propsed api indirectly, we don't need to have crafty the dts.
ie the dts files area manually created, with this approach we don't need to do that anymore, uts pulled down automatically

Comment (rchiodo) [reply package.json:25]: Forgot this dictates what .d.ts files are pulled down.

**Pull request merged**